src/assets/src/components/addQueue.tsx

```diff
@@ -0,0 +1,388 @@
+            </Alert>
+            {props.checkHostError ? <ErrorDisplay formErrors={[props.checkHostError]} /> : undefined}
+            <SingleInputField
+                id="add_host"
+                fieldComponent={StatelessInputGroupForm}
+                placeholder="Uniqname..."
+                buttonType='success'
+                onSubmit={(username) => {
+                    if (!hostUsernames.includes(username)) {
+                        props.onNewHost(username)
+                    }
+                }}
+                disabled={props.disabled}
+                fieldSchema={uniqnameSchema}
+                showRemaining={false}
+            >
+                + Add Host
+            </SingleInputField>
+            <h3>Current Hosts</h3>
+            <ListGroup>{hostsSoFar}</ListGroup>
+            <div className='mt-4'>
+                <Button variant='primary' aria-label='Back' disabled={props.disabled} onClick={props.onBack}>
+                    Back
+                </Button>
+                <Button
+                    variant='primary'
+                    className={buttonSpacing}
+                    aria-label='Finish Adding Queue'
+                    disabled={props.disabled}
+                    onClick={props.onFinish}
+                >
+                    Finish Adding Queue
+                </Button>
+                <CancelAddButton disabled={props.disabled} />
+            </div>
+        </div>
+    );
+}
+
+interface AddQueueEditorProps {
+    // Shared
+    disabled: boolean;
+    activeKey: string;
+    onTabSelect: (eventKey: string) => void;
+    // General Tab
+    name: string;
+    nameValidationResult?: ValidationResult;
+    onChangeName: (value: string) => void;
+    description: string;
+    descriptValidationResult?: ValidationResult;
+    onChangeDescription: (value: string) => void;
+    backends: {[backend_type: string]: string};
+    allowedMeetingTypes: Set<string>;
+    allowedIsInvalid?: boolean;
+    onChangeAllowed: (allowed: Set<string>) => void;
+    onGeneralNextClick: () => void;
+    // Manage Hosts Tab
+    currentUser?: User;
+    hosts: User[];
+    onNewHost: (username: string) => void;
+    checkHostError?: FormError;
+    onChangeHosts: (hosts: User[]) => void;
+    onBackClick: () => void;
+    onFinishClick: () => void;
+}
+
+// https://react-bootstrap.github.io/components/tabs/#tabs-custom-layout
+function AddQueueEditor(props: AddQueueEditorProps) {
+    return (
+        <Tab.Container id='add-queue-editor' defaultActiveKey='general' activeKey={props.activeKey} onSelect={props.onTabSelect}>
+            <Row>
+                <Col sm={3}>
+                    <Nav variant='pills' className='flex-column mt-5'>
+                        <Nav.Item><Nav.Link eventKey='general'>General</Nav.Link></Nav.Item>
+                        <Nav.Item><Nav.Link eventKey='hosts'>Manage Hosts</Nav.Link></Nav.Item>
+                    </Nav>
```

Comment:
@teshsopher, I was looking at how to make both of the Tab links keyboard accessible, but for some reason, the standard implementation of these components disallows that. You have to change the default `role` property from `tab` to something else, which may also be a problem in itself. I was looking at this section and trying to figure out what to do; it doesn't sound like they expect all links to be in the Tab order? https://www.w3.org/TR/wai-aria-practices/#tabpanel

I also implemented the "Next" and "Back" buttons at the bottom, so there should be an accessible way to navigate this. Let me know if you have any thoughts about the Tab links.